### ISSUE TITLE
Possible Changes

### ISSUE THREAD

**Issue by durandj** [macostea/cage#3]
I really like the idea of this project but I had a few things that I wanted to adjust about it but since they're relatively significant changes I wanted to ask before I worked on them. Let me know what you think and I'll work on PR's for the changes you feel comfortable with.

**Docker Image Selection**
Some users may want to select which Docker image they want their app to run in. This would be a nice little addition to give some more flexibility.

**Shell Command**
If you had a Docker container that you needed to make changes to it would be nice to be able to easily connect to the container. I know that in theory you could just use `docker ps` to find the container and then `docker exec -it <container> bash` to connect to it but it would be nice to have an easy to just say `cage app:shell` to connect to the container directly.

An example of why this would be useful is for dealing with Python libraries that require C libraries to be installed.

**Switch to Click**
[Click](http://click.pocoo.org/6/) is a command line parsing library that aims to be simpler and more powerful than `argparse`. It's a much nicer developer experience in my opinion.

**Comment by macostea** [comment]
These sound like really good ideas.
The last 2 seem lighter changes so you can open a PR for them whenever you like.
As for the first one I think we should discuss a bit more since I'm not sure how that would impact the users. If you already have a possible solution for it I'd love to hear more about it.

**Comment by durandj** [comment]
Yeah for sure. I'll spend some time putting some ideas together and make a separate issue to discuss the first change.
